### ISSUE TITLE
Issues with Leidenclustering

### ISSUE THREAD

**Issue by Intron7** [scverse/rapids_singlecell#44]
There is an issue with Leiden Clustering in cugraph version 23.06. 

The issue has been reported https://github.com/rapidsai/cugraph/issues/3749.

For the meantime please use rapids-23.04

**Comment by YH-Zheng** [comment]
Leiden clustering returned to normal after I created a new rapids-23.04 environment.

It greatly speeds up leiden clustering on large datasets, thanks for the great work!

**Comment by Intron7** [comment]
Rapids-23.08 still has this bug. If you need Leiden Clustering stay with Rapids-23.04

**Comment by Intron7** [comment]
I looks like leiden clustering will be fixed with rapids-23.12

**Comment by maestriev** [comment]
Just wanted to document another example of this leiden over-clustering. I used conda `env create -f conda/rsc_rapids_24.02.yml` which was in the installation instructions [here](https://rapids-singlecell.readthedocs.io/en/latest/Installation.html). The following is comparing the results for ~300,000 cells. 

Using the following rapids function:
```
rsc.tl.louvain(adata, resolution=0.6,key_added="louvain_0.6")
rsc.tl.louvain(adata, resolution=0.8,key_added="louvain_0.8")
```
Or the default:
```
sc.tl.louvain(adata, resolution=0.6,key_added="louvain_0.6_default")
sc.tl.louvain(adata, resolution=0.8,key_added="louvain_0.8_default")
```
<img width="910" alt="Screenshot 2024-05-17 at 11 58 25 PM" src="https://github.com/scverse/rapids_singlecell/assets/112463068/9a9a6b89-968d-47e1-a9f3-d565a97a635e">

I checked the above information and saw cugraph 24.02 and rapids 24.02. Looks like I will be trying with `env create -f conda/rsc_rapids_24.04.yml` tomorrow instead haha! So far the rest is working great
